tests/algorithms/test_EC.py

```diff
@@ -45,11 +45,68 @@ def test_invalid_algorithm(self):
         with pytest.raises(JWKError):
             ECDSAECKey({'kty': 'bla'}, ALGORITHMS.ES256)
 
+    def test_EC_jwk(self):
+        key = {
+            "kty": "EC",
+            "kid": "bilbo.baggins@hobbiton.example",
+            "use": "sig",
+            "crv": "P-521",
+            "x": "AHKZLLOsCOzz5cY97ewNUajB957y-C-U88c3v13nmGZx6sYl_oJXu9A5RkTKqjqvjyekWF-7ytDyRXYgCF5cj0Kt",
+            "y": "AdymlHvOiLxXkEhayXQnNCvDX4h9htZaCJN34kfmC6pV5OhQHiraVySsUdaQkAgDPrwQrJmbnX9cwlGfP-HqHZR1",
+            "d": "AAhRON2r9cqXX1hg-RoI6R1tX5p2rUAYdmpHZoC1XNM56KtscrX6zbKipQrCW9CGZH3T4ubpnoTKLDYJ_fF3_rJt",
+        }
+
+        assert not ECDSAECKey(key, ALGORITHMS.ES512).is_public()
+        assert not CryptographyECKey(key, ALGORITHMS.ES512).is_public()
+
+        del key['d']
+
+        # We are now dealing with a public key.
+        assert ECDSAECKey(key, ALGORITHMS.ES512).is_public()
+        assert CryptographyECKey(key, ALGORITHMS.ES512).is_public()
+
+        del key['x']
+
+        # This key is missing a required parameter.
+        with pytest.raises(JWKError):
+            ECDSAECKey(key, ALGORITHMS.ES512)
+
+        with pytest.raises(JWKError):
+            CryptographyECKey(key, ALGORITHMS.ES512)
+
     def test_verify(self):
         key = ECDSAECKey(private_key, ALGORITHMS.ES256)
         msg = b'test'
         signature = key.sign(msg)
         public_key = key.public_key()
 
         assert public_key.verify(msg, signature) == True
-        assert public_key.verify(msg, b'not a signature') == False
\ No newline at end of file
+        assert public_key.verify(msg, b'not a signature') == False
+
+    def assert_parameters(self, as_dict, private):
+        assert isinstance(as_dict, dict)
+
+        # Public parameters should always be there.
+        assert 'x' in as_dict
+        assert 'y' in as_dict
+        assert 'crv' in as_dict
+
+        assert 'kty' in as_dict
+        assert as_dict['kty'] == 'EC'
+
+        if private:
+            # Private parameters as well
+            assert 'd' in as_dict
+
+        else:
+            # Private parameters should be absent
+            assert 'd' not in as_dict
+
+    def test_to_dict(self):
```

Comment:
Split this into:
* `test_CryptographyECKey_public_to_dict`
* `test_CryptographyECKey_private_to_dict`
* `test_ ECDSAECKey_public_to_dict`
* `test_ ECDSAECKey_private_to_dict`